jitbuilder/release/src/Atomic.cpp

```diff
@@ -0,0 +1,136 @@
+/*******************************************************************************
+ *
+ * (c) Copyright IBM Corp. 2016, 2016
+ *
+ *  This program and the accompanying materials are made available
+ *  under the terms of the Eclipse Public License v1.0 and
+ *  Apache License v2.0 which accompanies this distribution.
+ *
+ *      The Eclipse Public License is available at
+ *      http://www.eclipse.org/legal/epl-v10.html
+ *
+ *      The Apache License v2.0 is available at
+ *      http://www.opensource.org/licenses/apache2.0.php
+ *
+ * Contributors:
+ *    Multiple authors (IBM Corp.) - initial implementation and documentation
+ ******************************************************************************/
+
+
+#include <iostream>
+#include <stdlib.h>
+#include <stdint.h>
+#include <dlfcn.h>
+#include <errno.h>
+
+#include "Jit.hpp"
+#include "ilgen/TypeDictionary.hpp"
+#include "ilgen/MethodBuilder.hpp"
+#include "Atomic.hpp"
+
+using std::cout;
+using std::cerr;
+
+
+int
+main(int argc, char *argv[])
+   {
+   cout << "Step 1: initialize JIT\n";
+   bool initialized = initializeJit();
+   if (!initialized)
+      {
+      cerr << "FAIL: could not initialize JIT\n";
+      exit(-1);
+      }
+
+   cout << "Step 2: define type dictionary\n";
+   TR::TypeDictionary types;
+
+   cout << "Step 3: compile two method builders\n";
+   AtomicMethod method(&types);
+   uint8_t *entry = 0;
+   int32_t rc = compileMethodBuilder(&method, &entry);
+   if (rc != 0)
+      {
+      cerr << "First method FAIL: compilation error " << rc << "\n";
+      exit(-2);
+      }
+
+   Atomic64Method method2(&types);
+   uint8_t *entry64 = 0;
+   int32_t rsc = compileMethodBuilder(&method2, &entry64);
+   if (rsc != 0)
+      {
+      cerr << "Second method FAIL: compilation error " << rsc << "\n";
+      exit(-2); 
+      }
+
+   cout << "Step 4: invoke compiled code and print results\n";
+   typedef int32_t (AtomicMethodFunction)(int32_t*);
+   AtomicMethodFunction *increment = (AtomicMethodFunction *) entry;
+   typedef int64_t (Atomic64MethodFunction)(int64_t*);
```

Comment:
Hi Mark, I renamed the files and function names to make it clear. (cpp/hpp named as AtomicOperations but exe became only atomic. It seems other testcases are also named in this way, only keeping the first half of the name) Still, I'm not sure about if the name I gave is appropriate. If not, please let me know. I will improve immediately.  Also, I inserted another step as you suggested. I think it looks better now.